server/startup/migrations/v225.js

```diff
@@ -0,0 +1,48 @@
+import { Migrations } from '../../../app/migrations';
+import { Settings, Users } from '../../../app/models/server';
+
+Migrations.add({
+	version: 225,
+	up() {
+		const hideAvatarsSetting = Settings.findById('Accounts_Default_User_Preferences_hideAvatars');
+		const hideAvatarsSidebarSetting = Settings.findById('Accounts_Default_User_Preferences_sidebarHideAvatar');
+		Settings.removeById('Accounts_Default_User_Preferences_sidebarShowDiscussion');
+		Settings.removeById('Accounts_Default_User_Preferences_sidebarHideAvatar');
+		Settings.upsert({
+			_id: 'Accounts_Default_User_Preferences_sidebarDisplayAvatar',
+		}, {
+			$set: {
+				value: !hideAvatarsSidebarSetting.value,
+			},
+		});
+		Settings.removeById('Accounts_Default_User_Preferences_hideAvatars');
+		Settings.upsert({
+			_id: 'Accounts_Default_User_Preferences_displayAvatars',
+		}, {
+			$set: {
+				value: !hideAvatarsSetting.value,
+			},
+		});
+		const users = Users.find({}, { fields: { _id: 1, 'settings.preferences': 1 } }).fetch();
+		users.forEach((user) => {
+			if (!(user.settings && user.settings.preferences)) {
+				return;
+			}
+			Users.update(
+				{ _id: user._id },
+				{
+					$set: {
+						'settings.preferences.displayAvatars': !user.settings.preferences.hideAvatars,
```

Comment:
it's possible for a user to not to have the `hideAvatars` in `preferences`? (because if so, this can crash)

Reply:
I don't think it's gonna crash, but it will have the wrong behaviour. It shouldn't be set if the value is undefined. I'll push a fix for this.